AL1012: 'arm64' is not a valid setting for option 'platform' thrown when localized resources are available and debugging on remote iOS device

---
**Issue moved from dotnet/maui#10342**
- Please respond to @ewerspej.

---

_From @ewerspej on Tuesday, September 27, 2022 10:30:08 AM_

### Description

When I try to debug an app on a remote iOS device (iPhone X running iOS 15.5) that is attached to my MacBook Pro, I keep receiving this error message in the console: 

AL1012: 'arm64' is not a valid setting for option 'platform'

**This only occurs when localized resources are available** (e.g. _resources.de.resx_)

### Steps to Reproduce

1. Use VS 17.4 Preview 2.1 for Windows
2. Create a .NET MAUI App project with .NET 6.0 or .NET 7.0
3. Add translations for strings, e.g. _resources.de.resx_
4. Attach an iPhone X running iOS 15.5 to your Mac
5. Connect to Mac
6. Start debugging on the remote device

Expected result: App runs on remote device
Actual result: AL1012: 'arm64' is not a valid setting for option 'platform'


### Link to public reproduction project repository

https://github.com/ewerspej/dotnet-maui-bug-repro-ios-arm64

### Version with bug

7.0.100-rc.1.22431.12
_(not working with previous version, either)_

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.5

### Did you find any workaround?

No

### Relevant log output

```shell
Build started...
1>------ Build started: Project: VisualCountDown, Configuration: Debug Any CPU ------
1>C:\Program Files\dotnet\sdk\7.0.100-rc.1.22431.12\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
1>Executing SayHello Task to establish a connection to a Remote Server. 
1>			Properties: 
1>				SessionId=98293d747ad47ec5798b143b8f1f5246fe28a7ddf0ca91ed602593262c492f64, 
1>				Addresss=192.168.30.24, 
1>				SshPort=22, 
1>				TcpPort=59112, 
1>				User=Julian Ewers-Peters, 
1>				AppName=VisualCountDown,
1>				VisualStudioProcessId=11908,
1>				ContinueOnDisconnected=False
1>Executing SayHello Task to establish a connection to a Remote Server. 
1>			Properties: 
1>				SessionId=98293d747ad47ec5798b143b8f1f5246fe28a7ddf0ca91ed602593262c492f64, 
1>				Addresss=192.168.30.24, 
1>				SshPort=22, 
1>				TcpPort=59112, 
1>				User=Julian Ewers-Peters, 
1>				AppName=VisualCountDown,
1>				VisualStudioProcessId=11908,
1>				ContinueOnDisconnected=False
1>Detected signing identity:
1>  Code Signing Key: "Apple Development: Julian Ewers-Peters (H4YNPVSAM5)" (54D2880D911F98DEECF603395F14A9C0A6AAFCE1)
1>  Provisioning Profile: "VS: WildCard Development" (1fa321df-dcb0-4c46-95ac-3ce18f82f3fc)
1>  Bundle Id: com.companyname.visualcountdown
1>  App Id: P9BACKA8J2.com.companyname.visualcountdown
1>Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers.
1>C:\Users\ewers\source\repos\VisualCountDown\VisualCountDown\Helpers\ServiceHelper.cs(5,25,5,26): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
1>ALINK : error AL1012: 'arm64' is not a valid setting for option 'platform'
1>Done building project "VisualCountDown.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Elapsed 00:10,664 ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========
========== Elapsed 00:10,664 ==========
```

--- a/dotnet/targets/Xamarin.Shared.Sdk.props
+++ b/dotnet/targets/Xamarin.Shared.Sdk.props
@@ -34,6 +34,11 @@
 
 		<!-- Do we need a RuntimeIdentifier? For apps and app extensions we do -->
 		<_RuntimeIdentifierIsRequired Condition="'$(OutputType)' == 'Exe' Or '$(IsAppExtension)' == 'true'">true</_RuntimeIdentifierIsRequired>
+
+		<!-- The logic to compute the default value for GenerateSatelliteAssembliesForCore is wrong when building inside Visual Studio (Windows), so make sure it defaults to the correct value -->
+		<!-- Ref: https://github.com/xamarin/xamarin-macios/issues/16193 -->
+		<!-- Ref: https://github.com/dotnet/sdk/issues/28419 -->
+		<GenerateSatelliteAssembliesForCore Condition="'$(GenerateSatelliteAssembliesForCore)' == ''">true</GenerateSatelliteAssembliesForCore>
 	</PropertyGroup>
 
 	<!-- Set the default RuntimeIdentifier if not already specified. -->

